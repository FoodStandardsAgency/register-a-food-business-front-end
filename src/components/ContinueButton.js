import { Button } from "govuk-react";
import { ButtonArrow } from "@govuk-react/icons";
import styled from "@emotion/styled";
import { withTranslation } from "../../i18n.js";

const StyledButton = styled(Button)`
  margin-bottom: 18px;
`;

const ContinueButton = (props) => (
  <StyledButton
    onClick={props.onClick}
    disabled={props.disabled}
    id="continue-button"
    type="submit"
    icon={props.type === "begin" ? <ButtonArrow /> : null}
    start={props.type === "begin"}
  >
    {props.type === "begin"
      ? props.t("Begin registration")
      : props.type === "submit"
      ? props.t("Submit")
      : props.editModeFirstPage
      ? props.t("Save and continue")
      : "Continue"}
  </StyledButton>
);

export default withTranslation("common")(ContinueButton);
