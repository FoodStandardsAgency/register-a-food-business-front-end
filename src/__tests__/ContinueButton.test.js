import ContinueButton from "../components/ContinueButton";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("<ContinueButton />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<ContinueButton />);
    expect(wrapper.length).toBe(1);
  });

  it("matches the previous snapshot", () => {
    const tree = renderer.create(<ContinueButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("is passed the 'type' prop", () => {
    const wrapper = mount(<ContinueButton type="test" />);
    const continueButton = wrapper.find("ContinueButton");
    expect(continueButton.props().type).toBe("test");
  });

  describe("when not given a type prop", () => {
    const wrapper = mount(<ContinueButton />);

    it("has 'Continue' as the button text", () => {
      expect(wrapper.text()).toContain("Continue");
    });
  });

  describe("given a type of 'begin'", () => {
    const wrapper = mount(<ContinueButton type="begin" />);
    it("has 'Begin registration' as the button text", () => {
      expect(wrapper.text()).toContain("Begin registration");
    });

    it("contains a <ButtonArrow> component", () => {
      const buttonArrow = wrapper.find("ButtonArrow");
      expect(buttonArrow.length).toBe(1);
    });
  });

  describe("given a type of 'submit'", () => {
    const wrapper = mount(<ContinueButton type="submit" />);

    it("has 'Submit' as the button text", () => {
      expect(wrapper.text()).toContain("Submit");
    });
  });
});
