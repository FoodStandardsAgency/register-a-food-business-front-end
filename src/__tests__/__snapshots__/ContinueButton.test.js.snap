// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContinueButton /> matches the previous snapshot 1`] = `
.emotion-0 {
  background-color: #00823b;
  border: none;
  box-shadow: 0 2px 0 #003618;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  font-size: 1em;
  line-height: 1.25;
  outline-offset: -1px;
  outline: 1px solid transparent;
  padding: .526315em .789473em .263157em;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-appearance: none;
  -webkit-font-smoothing: antialiased;
  opacity: 1;
  pointer-events: auto;
  padding-right: .84211em;
  margin-bottom: 15px;
}

.emotion-0:hover {
  background-color: #00692f;
  color: #ffffff;
}

.emotion-0:focus {
  color: #ffffff;
  background-color: #00692f;
  outline: 3px solid #ffbf47;
}

.emotion-0:active {
  position: relative;
  top: 2px;
  box-shadow: 0 0 0 #003618;
}

.emotion-0:visited {
  color: #00823b;
}

.emotion-0 svg {
  max-width: 15px;
  margin-left: 20px;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    margin-bottom: 20px;
  }
}

<button
  className="emotion-0"
  disabled={false}
  id="continue-button"
  type="submit"
>
  Continue
</button>
`;
